src/utilities/filetypes/EpwFile.cpp

```diff
@@ -567,79 +568,271 @@ boost::optional<EpwDesignCondition> EpwDesignCondition::fromDesignConditionsStri
-  dc.setCoolingEnthalpyMeanCoincidentDryBulb2(strings[EpwDesignField::CoolingEnthalpyMeanCoincidentDryBulb2]);
-  dc.setCoolingHours8To4AndDryBulb12pt8To20pt6(strings[EpwDesignField::CoolingHours8To4AndDryBulb12pt8To20pt6]);
-  dc.setExtremeWindSpeed1(strings[EpwDesignField::ExtremeWindSpeed1]);
-  dc.setExtremeWindSpeed2pt5(strings[EpwDesignField::ExtremeWindSpeed2pt5]);
-  dc.setExtremeWindSpeed5(strings[EpwDesignField::ExtremeWindSpeed5]);
-  dc.setExtremeMaxWetBulb(strings[EpwDesignField::ExtremeMaxWetBulb]);
-  dc.setExtremeMeanMinDryBulb(strings[EpwDesignField::ExtremeMeanMinDryBulb]);
-  dc.setExtremeMeanMaxDryBulb(strings[EpwDesignField::ExtremeMeanMaxDryBulb]);
-  dc.setExtremeStdDevMinDryBulb(strings[EpwDesignField::ExtremeStdDevMinDryBulb]);
-  dc.setExtremeStdDevMaxDryBulb(strings[EpwDesignField::ExtremeStdDevMaxDryBulb]);
-  dc.setExtremeN5YearsMinDryBulb(strings[EpwDesignField::ExtremeN5YearsMinDryBulb]);
-  dc.setExtremeN5YearsMaxDryBulb(strings[EpwDesignField::ExtremeN5YearsMaxDryBulb]);
-  dc.setExtremeN10YearsMinDryBulb(strings[EpwDesignField::ExtremeN10YearsMinDryBulb]);
-  dc.setExtremeN10YearsMaxDryBulb(strings[EpwDesignField::ExtremeN10YearsMaxDryBulb]);
-  dc.setExtremeN20YearsMinDryBulb(strings[EpwDesignField::ExtremeN20YearsMinDryBulb]);
-  dc.setExtremeN20YearsMaxDryBulb(strings[EpwDesignField::ExtremeN20YearsMaxDryBulb]);
-  dc.setExtremeN50YearsMinDryBulb(strings[EpwDesignField::ExtremeN50YearsMinDryBulb]);
-  dc.setExtremeN50YearsMaxDryBulb(strings[EpwDesignField::ExtremeN50YearsMaxDryBulb]);
-  return boost::optional<EpwDesignCondition>(dc);
+  const auto& title = strings[EpwDesignField::TitleOfDesignCondition];
+  static const boost::regex yearRegex(R"(\b(20\d{2})\b)");  // Matches 20xx
+  boost::optional<ASHRAEHoFVintage> assumedVersion;
+  boost::smatch m;
+  if (boost::regex_search(title, m, yearRegex)) {
+    int year = std::stoi(m[1]);
+    if (year >= 2009 && (year - 2009) % 4 == 0) {
+      // Valid ASHRAE design year: 2009, 2013, 2017, 2021, 2025, etc.
+      if (year == 2009) {
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2009;
+      } else if (year == 2013) {
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2013;
+      } else if (year == 2017) {
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2017;
+      } else if (year == 2021) {
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2021;
+      } else if (year == 2025) {
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2025;
+      } else {
+        // Assume future version is the same as 2025
+        LOG_FREE(Debug, "openstudio.EpwFile",
+                 "Unrecognized ASHRAE design year (" << year << ") in EPW design condition title. Assuming 2025 vintage for parsing.");
+        assumedVersion = ASHRAEHoFVintage::ASHRAE_2025;
+      }
+    } else {
+      LOG_FREE(Debug, "openstudio.EpwFile", "Did not detect a valid ASHRAE design year in EPW design condition title.");
+    }
+  }
```

Comment:
So, I start by trying to figure out the version in the title. Technically speaking, this isn't necessary, but I would like to Warn the user if there is a valid year source in there but the number of fields don't match: we look for `20xx` and if it's 2009 or any four (4) year increment, it's valid.